python/strings_udf/strings_udf/__init__.py

```diff
@@ -59,13 +64,33 @@ def compiler_from_ptx_file(path):
                 "files. Please file an issue reporting this error and how you "
                 "installed cudf and strings_udf."
             )
-        sms = [
-            os.path.basename(f).rstrip(".ptx").lstrip("shim_") for f in files
-        ]
-        selected_sm = max(sm for sm in sms if sm < cc)
-        ptxpath = os.path.join(
-            os.path.dirname(__file__), f"shim_{selected_sm}.ptx"
-        )
+
+        virtual_sms = []
+        native_sms = []
+        for f in files:
+            file_name = os.path.basename(f)
+            sm_number = int(
+                file_name.rstrip(".ptx").lstrip("shim_").rstrip("-real")
```

Comment:
You need to strip `-real` and `-virtual`

Reply:
It might be best to update the CMake side to transform `CMAKE_CUDA_ARCHITECTURES` to be a unique set of architectures without any `-real` and `-virtual` postfixes so the python loading side has a 1 to 1 mapping.